### ISSUE TITLE
mem leak 1.10

### ISSUE THREAD

**Issue by xpiwo** [oracle/node-oracledb#472]
looks like the RS mem leak is back.

could you add load tests to your release cycle?

**Comment by cjbj** [comment]
@xpiwo We do run a bunch of load & stress tests.  You'll need to give us a test case.

**Comment by xpiwo** [comment]
```
var load = function(){
    var wf = [],
        result_set,
        db_con;

    wf.push(db.get_connection);

    wf.push(function(con, acb){
        db_con = con;
        db_con.execute(`
            BEGIN
                open :cursor for select * from emp;
            END;
            `, [{type: oracledb.CURSOR, dir: oracledb.BIND_OUT}], acb);
    });

    wf.push(function(result, acb){
        result_set = result.outBinds[0];
        result_set.getRows(50, acb);
    });

    wf.push(function(rows, acb){
        // console.log(rows);
        acb(false);
    });

    async.waterfall(wf, function(err){
        if(err)
            console.log('ignore errors for brevity');

        result_set.close(function(){
            db_con.release(function(){
                setTimeout(load, 1);
            });
        });

    });
};

for(var i=0;i<30;i++)
    load();
```

```
1.10.0:
  PID USER      PR  NI  VIRT  RES  SHR S %CPU %MEM    TIME+  COMMAND
 4642 root      20   0 1390m 305m  20m R 189.6 16.3  10:01.99 node

1.9.3:
  PID USER      PR  NI  VIRT  RES  SHR S %CPU %MEM    TIME+  COMMAND
 5129 root      20   0 1391m  98m  20m R 184.6  5.3  10:04.67 node
```

**Comment by cjbj** [comment]
Is 'emp' supposed to exist?  Why are you ignoring errors?

**Comment by xpiwo** [comment]
it's a simplified test case, there are no errors.

https://apexplained.wordpress.com/2013/04/20/the-emp-and-dept-tables-in-oracle/

**Comment by cjbj** [comment]
@xpiwo thanks. Looks like this is specific to REF CURSORS.  We'll work on a fix and continue to improve testing.

**Comment by cjbj** [comment]
@xpiwo Fixed in 1.10.1.  Thanks for reporting it!
